includes/Modules/Sign_In_With_Google.php

```diff
@@ -36,12 +44,141 @@ final class Sign_In_With_Google extends Module implements Module_With_Assets, Mo
 	 */
 	const MODULE_SLUG = 'sign-in-with-google';
 
+	/**
+	 * Validate_Auth_Request instance.
+	 *
+	 * @since n.e.x.t
+	 * @var Validate_Auth_Request
+	 */
+	protected $validate_auth_request;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Context        $context        Plugin context.
+	 * @param Options        $options        Optional. Option API instance. Default is a new instance.
+	 * @param User_Options   $user_options   Optional. User Option API instance. Default is a new instance.
+	 * @param Authentication $authentication Optional. Authentication instance. Default is a new instance.
+	 * @param Assets         $assets         Optional. Assets API instance. Default is a new instance.
+	 */
+	public function __construct(
+		Context $context,
+		Options $options = null,
+		User_Options $user_options = null,
+		Authentication $authentication = null,
+		Assets $assets = null
+	) {
+		parent::__construct( $context, $options, $user_options, $authentication, $assets );
+
+		$this->validate_auth_request = new Validate_Auth_Request( $context );
+	}
+
 	/**
 	 * Registers functionality through WordPress hooks.
 	 *
 	 * @since 1.137.0
 	 */
 	public function register() {
+		add_action(
+			'init',
+			function () {
+				add_rewrite_rule( '^auth/google/?$', 'index.php?google_auth=1', 'top' );
+			}
+		);
+
+		add_filter(
+			'query_vars',
+			function ( $vars ) {
+				$vars[] = 'google_auth';
+				return $vars;
+			}
+		);
```

Comment:
Thanks, @zutigrm, but this is not going to work. We need to rework it to use the `login_form_{$action}` action to process the sign in with google action. It means that we should use the login URL with our own action as the redirect URI where we will handle results and show the error if something is wrong, or log in the user on success.

https://github.com/WordPress/WordPress/blob/master/wp-login.php#L571

Reply:
@zutigrm that "endpoint" is used to process the result of the OAuth flow that signs in users on the Google side. The suggested approach in AC is not going to work, we must use a login form action for that which is intended exactly for this kind of things.

Reply:
It won't work, because it is not sufficient to just add a new rewrite rule, we must flush all existing rewrite rules in the database which is not a simple thing since we can't do it on every request.

Reply:
Anyway, using custom rewrite rules for what we need here is a bad design because WordPress already has login "API" that should be used for that.

Reply:
@eugene-manuilov Yeah, I was thinking where would be best place to flush rewrite rules, once on module setup perhaps would be sufficient. But changing this to login form url does make more sense/less potential complications for the users. I will switch the logic. Thanks